From cdef313bd34c5abd897b80f25554b0c66737ed05 Mon Sep 17 00:00:00 2001
From: Kai Kang <kai.kang@windriver.com>
Date: Tue, 14 Oct 2025 01:37:35 +0000
Subject: [PATCH] There are conflict of config files between
 kea and lib32-kea:

| Error: Transaction test error:
|  file /etc/kea/kea-ctrl-agent.conf conflicts between attempted installs of
     lib32-kea-1.7.10-r0.core2_32 and kea-1.7.10-r0.core2_64
|  file /etc/kea/kea-dhcp4.conf conflicts between attempted installs of
     lib32-kea-1.7.10-r0.core2_32 and kea-1.7.10-r0.core2_64
|  file /etc/kea/kea-dhcp6.conf conflicts between attempted installs of
     lib32-kea-2.6.1-r0.core2_32 and kea-2.6.1-r0.core2_64

Because they are all commented out, replace the expanded libdir path with
'$libdir' in the config files to avoid conflict.

Upstream-Status: Submitted [https://gitlab.isc.org/isc-projects/kea/-/issues/2602]
Signed-off-by: Kai Kang <kai.kang@windriver.com>
Signed-off-by: Lei Maohui <leimaohui@fujitsu.com>
Signed-off-by: Liu Yiding <liuyd.fnst@fujitsu.com>
---
 src/bin/keactrl/kea-ctrl-agent.conf.pre | 3 ++-
 src/bin/keactrl/kea-dhcp4.conf.pre      | 6 +++---
 src/bin/keactrl/kea-dhcp6.conf.pre      | 6 +++---
 3 files changed, 8 insertions(+), 7 deletions(-)

diff --git a/src/bin/keactrl/kea-ctrl-agent.conf.pre b/src/bin/keactrl/kea-ctrl-agent.conf.pre
index 29d8111..de71f41 100644
--- a/src/bin/keactrl/kea-ctrl-agent.conf.pre
+++ b/src/bin/keactrl/kea-ctrl-agent.conf.pre
@@ -85,7 +85,8 @@
     // Agent will fail to start.
     "hooks-libraries": [
 //  {
-//      "library": "@libdir@/kea/hooks/control-agent-commands.so",
+//      // Replace $libdir with real library path /usr/lib or /usr/lib64
+//      "library": "$libdir/kea/hooks/control-agent-commands.so",
 //      "parameters": {
 //          "param1": "foo"
 //      }
diff --git a/src/bin/keactrl/kea-dhcp4.conf.pre b/src/bin/keactrl/kea-dhcp4.conf.pre
index 2a58507..86b5abf 100644
--- a/src/bin/keactrl/kea-dhcp4.conf.pre
+++ b/src/bin/keactrl/kea-dhcp4.conf.pre
@@ -255,7 +255,7 @@
     //       // of all devices serviced by Kea, including their identifiers
     //       // (like MAC address), their location in the network, times
     //       // when they were active etc.
-    //       "library": "@libdir@/kea/hooks/libdhcp_legal_log.so",
+    //       "library": "$libdir/kea/hooks/libdhcp_legal_log.so",
     //       "parameters": {
     //           "base-name": "kea-forensic4"
     //       }
@@ -271,14 +271,14 @@
     //       // of specific options or perhaps even a combination of several
     //       // options and fields to uniquely identify a client. Those scenarios
     //       // are addressed by the Flexible Identifiers hook application.
-    //       "library": "@libdir@/kea/hooks/libdhcp_flex_id.so",
+    //       "library": "$libdir/kea/hooks/libdhcp_flex_id.so",
     //       "parameters": {
     //           "identifier-expression": "relay4[2].hex"
     //       }
     //   },
     //   {
     //       // the MySQL host backend hook library required for host storage.
-    //       "library": "@libdir@/kea/hooks/libdhcp_mysql.so"
+    //       "library": "$libdir/kea/hooks/libdhcp_mysql.so"
     //   }
     // ],
 
diff --git a/src/bin/keactrl/kea-dhcp6.conf.pre b/src/bin/keactrl/kea-dhcp6.conf.pre
index c69a508..2bb488f 100644
--- a/src/bin/keactrl/kea-dhcp6.conf.pre
+++ b/src/bin/keactrl/kea-dhcp6.conf.pre
@@ -201,7 +201,7 @@
     //       // of all devices serviced by Kea, including their identifiers
     //       // (like MAC address), their location in the network, times
     //       // when they were active etc.
-    //       "library": "@libdir@/kea/hooks/libdhcp_legal_log.so",
+    //       "library": "$libdir/kea/hooks/libdhcp_legal_log.so",
     //       "parameters": {
     //           "base-name": "kea-forensic6"
     //       }
@@ -217,14 +217,14 @@
     //       // of specific options or perhaps even a combination of several
     //       // options and fields to uniquely identify a client. Those scenarios
     //       // are addressed by the Flexible Identifiers hook application.
-    //       "library": "@libdir@/kea/hooks/libdhcp_flex_id.so",
+    //       "library": "$libdir/kea/hooks/libdhcp_flex_id.so",
     //       "parameters": {
     //           "identifier-expression": "relay6[0].option[37].hex"
     //       }
     //   },
     //   {
     //       // the MySQL host backend hook library required for host storage.
-    //       "library": "@libdir@/kea/hooks/libdhcp_mysql.so"
+    //       "library": "$libdir/kea/hooks/libdhcp_mysql.so"
     //   }
     // ],
 
-- 
2.43.0

